src/PatientChecking/PatientChecking/Services/AppointmentService.cs

```diff
@@ -24,12 +25,67 @@ public async Task<AppointmentDashboard> GetAppointmentSummary()
+            var numberOfPatientsInMonth = await _patientCheckInContext.Appointments.Where(x => x.CheckInDate.Year == DateTime.Now.Year && x.CheckInDate.Month == DateTime.Now.Month)
+                                                                 .GroupBy(p => p.PatientId)
+                                                                 .Select(g => new { g.Key, count = g.Count() }).ToListAsync();
             return new AppointmentDashboard()
             {
                 NumOfAppointments = numberOfAppointments.Count(),
                 NumOfAppointmentsInMonth = numberOfAppointmentsInMonth.Count(),
                 NumOfAppointmentsInToday = numberOfAppointmentsInToday.Count(),
+                NumOfPatientsInMonth = numberOfPatientsInMonth.Count()
             };
         }
+
+        public async Task<AppointmentList> GetListAppoinmentsPaging(PagingRequest request)
+        {
+            var query = from appointment in _patientCheckInContext.Appointments
+                        join patient in _patientCheckInContext.Patients on appointment.Patient.PatientId equals patient.PatientId
+                        select new { appointment, patient };
+             if(request.SortSelection == 0)
+            {
+                query = query.OrderBy(i => i.patient.FullName);
+            }
+            else if(request.SortSelection == 1)
+            {
+                query = query.OrderBy(i => i.patient.PatientIdentifier); 
+            }
+            else if(request.SortSelection == 2)
+            {
+                query = query.OrderBy(i => i.patient.DoB);
+            }
+            else if (request.SortSelection == 3)
+            {
+                query = query.OrderByDescending(i => i.appointment.CheckInDate);
+            }
+            else
+            {
+                query = query.OrderBy(i => i.appointment.Status == AppointmentStatus.CheckIn.ToString() ? 1 
+                                         : i.appointment.Status == AppointmentStatus.Closed.ToString() ? 2
+                                         : 3
+                );
+            }
+            int totalRow = query.Count();
+            var data = query.Skip((request.PageIndex - 1) * request.PageSize)
+                .Take(request.PageSize).Select(x => new ServiceModels.Appointment()
+                {
+                    AppointmentId = x.appointment.AppointmentId,
+                    CheckInDate = x.appointment.CheckInDate,
+                    Status = x.appointment.Status,
+                    Patient = new ServiceModels.Patient()
+                    {
+                        AvatarLink = x.patient.AvatarLink,
+                        DoB = x.patient.DoB,
+                        FullName = x.patient.FullName,
+                        PatientIdentifier = x.patient.PatientIdentifier
+                    }
+                }).ToList();
+            AppointmentList appointmentList = new()
```

Comment:
This syntax is only available on C# 9.0 or higher. Please use syntax `AppointmentList appointmentList = new AppointmentList()`

Reply:
AppointmentList appointmentList = new AppointmentList() has been applied in code